src/query/query_parser/query_grammar.rs

```diff
@@ -83,28 +83,69 @@ parser! {
 }
 
 parser! {
+    /// Function that parses a range out of a Stream
+    /// Supports ranges like:
+    /// [5 TO 10], {5 TO 10}, [* TO 10], [10 TO *], {10 TO *], >5, <=10
+    /// [a TO *], [a TO c], [abc TO bcd}
     fn range[I]()(I) -> UserInputLeaf
     where [I: Stream<Item = char>] {
         let range_term_val = || {
             word().or(negative_number()).or(char('*').with(value("*".to_string())))
         };
+
+        // check for unbounded range in the form of <5, <=10, >5, >=5
+        let elastic_unbounded_range = (choice([attempt(string(">=")),
+                                               attempt(string("<=")),
+                                               attempt(string("<")),
+                                               attempt(string(">"))])
+                                       .skip(spaces()),
+                                       range_term_val()).
+            map(|(comparison_sign, bound): (&str, String)|
+                match comparison_sign {
+                    ">=" => return (UserInputBound::Inclusive(bound), UserInputBound::Unbounded),
+                    "<=" => return (UserInputBound::Unbounded, UserInputBound::Inclusive(bound)),
+                    "<" => return (UserInputBound::Unbounded, UserInputBound::Exclusive(bound)),
+                    ">" => return (UserInputBound::Exclusive(bound), UserInputBound::Unbounded),
+                    // default case
+                    _ => return (UserInputBound::Unbounded, UserInputBound::Unbounded)
+                });
         let lower_bound = (one_of("{[".chars()), range_term_val())
             .map(|(boundary_char, lower_bound): (char, String)|
-                if boundary_char == '{' { UserInputBound::Exclusive(lower_bound) }
-                else { UserInputBound::Inclusive(lower_bound) });
+                 if lower_bound == "*" {
+                     UserInputBound::Unbounded
+                 } else {
+                     if boundary_char == '{' {
+                         UserInputBound::Exclusive(lower_bound)
+                     } else {
+                         UserInputBound::Inclusive(lower_bound)
+                     }});
```

Comment:
i am not fond of the position of the second `}`.

Reply:
I see your point, moved the last "}" of both such if-elses to respective new lines